crates/payload/basic/src/lib.rs

```diff
@@ -635,6 +638,10 @@ where
 {
     let BuildArguments { client, pool, mut cached_reads, config, cancel, best_payload } = args;
 
+    let state_provider = client.state_by_block_hash(config.parent_block.hash)?;
+    let state: State<&StateProviderBox<'_>> = State::new(&state_provider);
+    let wrapped_state = WrapDatabaseRef(cached_reads.as_db(&state));
+    let mut db = RevmStateBuilder::default().with_database(Box::new(wrapped_state)).build();
```

Comment:
any reason for this type?

(in general we have a few too many types wrapped around each other, would be great if we could simplify)

Reply:
Talked in the call